sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java

```diff
@@ -44,58 +44,45 @@ private MarathonConstraintParser() {
     }
 
     /**
-     * ANDs the provided marathon-style constraint string onto the provided hard-coded
-     * {@link PlacementRule}, or returns the provided {@link PlacementRule} as-is if the
-     * marathon-style constraint is {@code null} or empty.
-     *
-     * @throws IOException if {@code marathonConstraints} couldn't be parsed, or if the parsed
-     *     content isn't valid or supported
-     */
-    public static PlacementRule parseWith(PlacementRule rule, String marathonConstraints)
-            throws IOException {
-        PlacementRule marathonRule = parse(marathonConstraints);
-        if (marathonRule instanceof PassthroughRule) {
-            return rule; // pass-through original rule
-        }
-        return new AndRule(rule, marathonRule);
-    }
-
-    /**
      * Creates and returns a new {@link PlacementRule} against the provided marathon-style
      * constraint string. Returns a {@link PassthroughRule} if the provided constraint string is
      * {@code null} or empty.
      *
+     *
+     * @param podName
      * @param marathonConstraints the marathon-style constraint string, containing one or more
      *     nested json list entries of the form {@code [["multi","list","value"],["hello","hi"]]},
      *     or one or more colon-separated entries of the form {@code multi:list:value,hello:hi},
      *     or a {@code null} or empty value if no constraint is defined
      * @throws IOException if {@code marathonConstraints} couldn't be parsed, or if the parsed
      *     content isn't valid or supported
      */
-    public static PlacementRule parse(String marathonConstraints) throws IOException {
+    public static PlacementRule parse(String podName, String marathonConstraints) throws IOException {
         if (marathonConstraints == null || marathonConstraints.isEmpty()) {
             // nothing to enforce
             return new PassthroughRule();
         }
         List<List<String>> rows = splitConstraints(marathonConstraints);
         if (rows.size() == 1) {
```

Comment:
Looks like downstream either turns podName into a StringMatcher, or ignores it, depending on the operation. How about just creating the StringMatcher here, and passing that to parseRow (and beyond) instead?